helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java

```diff
@@ -142,9 +142,11 @@ public synchronized void refresh(HelixDataAccessor accessor) {
     if (LOG.isDebugEnabled()) {
       LOG.debug("CurrentStates: " + _currentStateCache);
       LOG.debug("TargetExternalViews: " + _targetExternalViewCache.getExternalViewMap());
-      for (String customizedStateType : _sourceDataTypeMap.get(PropertyType.CUSTOMIZEDVIEW)) {
-        LOG.debug("CustomizedViews customizedStateType: " + customizedStateType + " "
-            + _customizedViewCaches.get(customizedStateType).getCustomizedViewMap());
+      if (_sourceDataTypeMap.containsKey(PropertyType.CUSTOMIZEDVIEW)) {
```

Comment:
getOrDefault(PropertyType.CUSTOMIZEDVIEW, Collections.emptyMap()) would be easier.

Reply:
@jiajunwang Addressed. Could you please confirm? Then we can merge it.